server/src/main/scala/Main.scala

```diff
@@ -19,16 +20,19 @@ class ExampleImplementation extends GreeterFs2Grpc[IO] {
   }
 }
 
-object Main extends StreamApp[IO] {
+object Main extends IOApp {
   val helloService: ServerServiceDefinition =
     GreeterFs2Grpc.bindService(new ExampleImplementation)
-  def stream(args: List[String], requestShutdown: IO[Unit]): fs2.Stream[IO, StreamApp.ExitCode] = {
+  def run(args: List[String]): IO[ExitCode] = {
     ServerBuilder
       .forPort(9999)
       .addService(helloService)
       .addService(ProtoReflectionService.newInstance())
       .stream[IO]
       .evalMap(server => IO(server.start()))
       .evalMap(_ => IO.never)
+      .compile
```

Comment:
WDYT of this:

```scala
  val serverResource: Resource[IO, Server] =
    ServerBuilder
      .forPort(9999)
      .addService(helloService)
      .addService(ProtoReflectionService.newInstance())
      .resource[IO]

  def run(args: List[String]): IO[ExitCode] = {
    serverResource.use { server =>
      IO(server.start()) *> IO.never.as(ExitCode.Success)
    }
  }
```

It is more symmetric with the client code, and removes an unnecessary stream.

Reply:
Good idea.  Updated.